lib/timeout.js

```diff
@@ -43,33 +43,32 @@ import wrapAsync from './internal/wrapAsync';
  * });
  */
 export default function timeout(asyncFn, milliseconds, info) {
-    var originalCallback, timer;
-    var timedOut = false;
+    var fn = wrapAsync(asyncFn);
 
-    function injectedCallback() {
-        if (!timedOut) {
-            originalCallback.apply(null, arguments);
-            clearTimeout(timer);
-        }
-    }
+    return initialParams(function (args, callback) {
+        var timedOut = false;
+        var timer;
 
-    function timeoutCallback() {
-        var name = asyncFn.name || 'anonymous';
-        var error  = new Error('Callback function "' + name + '" timed out.');
-        error.code = 'ETIMEDOUT';
-        if (info) {
-            error.info = info;
+        function timeoutCallback() {
+            var name = asyncFn.name || 'anonymous';
+            var error  = new Error('Callback function "' + name + '" timed out.');
+            error.code = 'ETIMEDOUT';
+            if (info) {
+                error.info = info;
+            }
+            timedOut = true;
+            callback(error);
         }
-        timedOut = true;
-        originalCallback(error);
-    }
 
-    var fn = wrapAsync(asyncFn);
+        args.push(function () {
```

Comment:
This is fine.  `args.concat(namedFunction)` is nice because it's a singe expression that returns the resulti, but it also creates a new array.  `push` is fine here because nothing else re-uses the `args` array.